base/compiler/typelattice.jl

```diff
@@ -122,6 +122,8 @@ function ⊑(@nospecialize(a), @nospecialize(b))
     (a === Any || b === NOT_FOUND) && return false
     a === Union{} && return true
     b === Union{} && return false
+    @assert !isa(a, TypeVar) "invalid lattice item"
```

Comment:
This function is somewhat hot. Move this check all the way down to the `===` check?

Reply:
I don't think a pointer comparison will affect this, relative to the many other egal and subtyping test it does